src/connector/src/source/pulsar/source/reader.rs

```diff
@@ -444,7 +444,7 @@ impl PulsarIcebergReader {
         if let Some(secret_key) = &self.props.aws_auth_props.secret_key {
             iceberg_configs.insert(
                 "iceberg.table.io.secret_access_key".to_string(),
-                secret_key.to_string(),
+                secret_key.expose_secret().to_string(),
```

Comment:
This actually shows that why calling `to_string` on `&String` or `&str` is bad: the method exists on all types that implements `Display`, and changing the type of `secret_key` will not always lead to compile error. We should prefer `to_owned` in such cases.

Reply:
Oh, it also works by not implementing `Display` on the new type `SecretString`. 😄